youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,136 @@
+            'id': 'tJluaH4BJ3v',
+            'title': 'Vidlii is against me',
+            'description': 'I have HAD it. Vidlii does not like me. I have tried to uplaod videos and submit them to the '
+                           'contest and no ne of my videos show up so maybe it is broken for everyone else but this one was '
+                           'trying to submit it because I wanted to submit to the contest :) Tanks I hope the website is '
+                           'fixed PS: Jan you are cool please add my video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/tJluaH4BJ3v.jpg',
+            'uploader': 'APPle5auc31995',
+            'url': 'https://cdn.vidlii.com/videos/tJluaH4BJ3v.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/APPle5auc31995',
+            'upload_date': '20171107',
+            'categories': 'News & Politics',
+            'tags': ['Vidlii', 'Jan', 'Videogames'],
+            'duration': 212,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.vidlii.com/watch?v=vBo2IcrwOkO',
+        'md5': 'b42640a596b4dc986702567d49268963',
+        'info_dict': {
+            'id': 'vBo2IcrwOkO',
+            'title': '(OLD VIDEO) i like youtube!!',
+            'description': 'Original upload date:<br />\nMarch 10th 2011<br />\nCredit goes to people who own content in the video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/vBo2IcrwOkO.jpg',
+            'uploader': 'MyEditedVideoSpartan',
+            'url': 'https://cdn.vidlii.com/videos/vBo2IcrwOkO.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/MyEditedVideoSpartan',
+            'upload_date': '20171011',
+            'categories': 'Film & Animation',
+            'tags': None,
+            'duration': 34,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # extract videoInfo variable for further use
+        videoInfo_dict = self._parse_json(js_to_json(self._html_search_regex(r'var videoInfo\s*=\s*({[^}]*})', webpage,
+                                                                             'videoInfo', fatal=False)), video_id)
+
+        # extract basic properties of video
+        title = (self._html_search_regex(r'<title>([^<]+?)</title>', webpage, 'title', default='') or
+                 self._html_search_meta('twitter:title', webpage, 'title', default='')).replace(" - VidLii",
+                                                                                                "") or self._html_search_regex(
+            r'<h1>(.+?)</h1>', webpage, 'title', default=None)
```

Comment:
I've already pointed out - title **must be fatal**.

Reply:
As far as I understood, setting *fatal=True* raises an Exception if the parameter could not be extracted. 

Above you can see something like ``title = get_from_source1 or get_from_source1 or get_from_source1``. If I set *fatal=True* for ``get_from_source1`` the extraction would stop if the title cannot be extracted, thus fallbacks ``get_from_source2`` and ``get_from_source3`` would never be used. How should this be handled? Should I check for a valid value in vidlii class itself?

Reply:
You must make fatal the last source.